app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,36 @@
+class ExpensesController < ApplicationController
+  
+  def create
+    begin
+      create_params = params.require(:expense).permit([:name, :amount, :due_date, :debtor_id, :hideout_id, :comments])
+      expense = Expense.create!(create_params)
+      render status: 201, json: expense.to_json
+    rescue ActionController::ParameterMissing, ActiveModel::StrictValidationFailed
+      render status: 400
+    end
+  end
+
+  def destroy
+    begin
+      expense = Expense.find!(params[:id])
+      expense.destroy
+      render status: 200
+    rescue ActiveRecord::RecordNotFound
+      render status: 404
+    end
+  end
+
+  def update
+    begin
+      update_params = params.require(:expense).permit([:name, :amount, :due_date, :debtor_id, :hideout_id, :comments])
+      expense = Expense.find!(params[:id])    
+      expense.update!(expense_params)
```

Comment:
Should not be updating hideout_id for this. I would frown upon using the `permit` function. For starters, its much less readable than simply requiring the attributes you need in a regular `permit.require` call by passing in an array. Also, with stuff like `comments` it becomes an issue since comments should be allowed to be `nil` since they can be blank but requiring it will cause rails to throw an error.

Reply:
This should not throw an error when the comment is `nil`, because permit just checks for permissible attributes. The `nil` checking is done in the model. I am still changing it to be consistent with the other controllers